### DISCUSSION TITLE
Debug logging for test containers

### DISCUSSION THREAD

**Discussion by ErikEngerd** [testcontainers/testcontainers-go#2901] [Q&A]
Is there a way to show detailed request/response logging for the API calls of test containers to docker?

**Comment by stevenh** [comment]
Hey @ErikEngerd what problem are you looking to solve with this?

**Reply by ErikEngerd** [reply]
I was debugging an issue where apparently there were some invalid characters in the response from the docker server (kubedock in this case). I am working on kubedock-dns  
 (https://github.com/ErikEngerd/kubedock-dns) to make testcontainers with kubedock on kubernetes easier to use. Kubedock implements a subset of the docker API for use with testcontainers. Kubedock-dns allows multiple concurrent tests to be run in the same namespace without interference and alleviates some current restrictions with kubedock. 

I managed to find the issue now by using an  http proxy that logged all requests (similar to tcpdump, but a bit more friendly). Using that I managed to solve my problem, but it would still be good to have a feature where one could enable detailed request response logging for testcontainers. That would have been easier since I would have seen the message in error more easily. Now I had to guess somewhat to find the offending message.
